### PR TITLE
feat(api) new endpoint: AddCommand

### PR DESCRIPTION
**Pull request by a-launois**
## Description

New endpoint:
```
POST /configuration/commands
```

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandRequest.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      * `3` - MISCELLANEOUS
+      * `4` - DISCOVERY
+    example: 2
+  command_line:
+    type: string
+    description: "Command line to be executed by the poller"
+  is_shell:
+    type: boolean
+    description: |
+      If your command requires shell features like pipes, redirections, globbing etc. check this box.
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandRequest.yaml:24]:
```suggestion
      If your command requires shell features like pipes, redirections, globbing etc., check this box.
```

File: centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandRequest.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+      If you are using the monitoring engine this option cannot be disabled.
+
+      Note that commands that require shell are slowing down the poller server.
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandRequest.yaml:28]:
```suggestion
      Note that commands that require the shell are slowing down the poller server.
```

File: centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandRequest.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    example: true
+  argument_example:
+    type: string
+    nullable: true
+    description: "Example of command argument's values"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandRequest.yaml:33]:
```suggestion
    description: "Example of command argument values"
```

File: centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandRequest.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        description:
+          type: string
+          nullable: true
+          description: "Description of the macro"
+          example: "macro-description"
+  connector_id:
+    type: integer
+    nullable: true
+    description: |
+      A connector is run in the background and execute specific commands without the need to execute a binary.
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandRequest.yaml:77]:
```suggestion
      A connector is run in the background and executes specific commands without the need to execute a binary.
```

File: centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandRequest.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  graph_template_id:
+    type: integer
+    nullable: true
+    description: "Graph template of the command"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandRequest.yaml:81]:
```suggestion
    description: "Graph template for the command"
```

File: centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandResponse.yaml [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+      * `3` - MISCELLANEOUS
+      * `4` - DISCOVERY
+    example: 2
+  command_line:
+    type: string
+    description: "Command line to be executed by the poller"
+  is_shell:
+    type: boolean
+    description: |
+      If your command requires shell features like pipes, redirections, globbing etc. check this box.
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandResponse.yaml:27]:
```suggestion
      If your command requires shell features like pipes, redirections, globbing etc., check this box.
```

File: centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandResponse.yaml [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+      If you are using the monitoring engine this option cannot be disabled.
+
+      Note that commands that require shell are slowing down the poller server.
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandResponse.yaml:31]:
```suggestion
      Note that commands that require the shell are slowing down the poller server.
```

File: centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandResponse.yaml [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+    description: "Indicates whether the command can be updated or not"
+    example: false
+  is_activated:
+    type: boolean
+    description: "Indicates whether the command is activated or not"
+    example: true
+  argument_example:
+    type: string
+    nullable: true
+    description: "Example of command argument's values"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandResponse.yaml:44]:
```suggestion
    description: "Example of command argument values"
```

File: centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandResponse.yaml [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+        description:
+          type: string
+          nullable: true
+          description: "Description of the macro"
+          example: "macro-description"
+  connector:
+    type: object
+    nullable: true
+    description: |
+      A connector is run in background and execute specific commands without the need to execute a binary.
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandResponse.yaml:88]:
```suggestion
      A connector is run in background and executes specific commands without the need to execute a binary.
```

File: centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandResponse.yaml [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+       description: "Connector ID"
+       example: 12
+      name:
+        type: "string"
+        description: "Connector name"
+        example: "SSH Connector"
+  graph_template:
+    type: object
+    nullable: true
+    description: "Graph template of the command"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandResponse.yaml:101]:
```suggestion
    description: "Graph template for the command"
```

File: centreon/doc/API/v23.10/Configuration/Command/Schema/FindCommandsResponse.yaml [hunk 11]
```diff
@@ -25,9 +25,9 @@ properties:
     type: boolean
     description: |
       If your command requires shell features like pipes, redirections, globbing etc. check this box.
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Command/Schema/FindCommandsResponse.yaml:27]:
```suggestion
      If your command requires shell features like pipes, redirections, globbing etc., check this box.
```

File: centreon/doc/API/v23.10/Configuration/Command/Schema/FindCommandsResponse.yaml [hunk 12]
```diff
@@ -25,9 +25,9 @@ properties:
...
-      
+
       If you are using the monitoring engine this option cannot be disabled.
-      
+
       Note that commands that require shell are slowing down the poller server.
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Command/Schema/FindCommandsResponse.yaml:31]:
```suggestion
      Note that commands that require the shell are slowing down the poller server.
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 13]
```diff
@@ -18273,3 +18273,23 @@ msgstr "Vous n'êtes pas autorisé à lister les hôtes"
 msgid "Error while searching for host configurations"
 msgstr "Erreur lors de la recherche des configurations d'hôtes"
 
+msgid "'%s' command name already exists"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18276]:
```suggestion
msgid "The '%s' command name already exists"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 14]
```diff
@@ -18273,3 +18273,23 @@ msgstr "Vous n'êtes pas autorisé à lister les hôtes"
...
+msgstr "Le nom de commande '%s' existe déjà"
+
+msgid "'%d' is not a valid command type"
+msgstr "'%d' n'est pas un type de commande valide"
+
+msgid "Following arguments are not valid: %s"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18282]:
```suggestion
msgid "The following arguments are not valid: %s"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 15]
```diff
@@ -18273,3 +18273,23 @@ msgstr "Vous n'êtes pas autorisé à lister les hôtes"
...
+msgstr "Les arguments suivante ne sont pas valides: %s"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18283]:
```suggestion
msgstr "Les arguments suivants ne sont pas valides: %s"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 16]
```diff
@@ -18273,3 +18273,23 @@ msgstr "Vous n'êtes pas autorisé à lister les hôtes"
...
+
+msgid "Following macros are not valid: %s"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18285]:
```suggestion
msgid "The following macros are not valid: %s"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 17]
```diff
@@ -18273,3 +18273,23 @@ msgstr "Vous n'êtes pas autorisé à lister les hôtes"
...
+msgstr "Les macros suivnates ne sont pas valides: %s"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18286]:
```suggestion
msgstr "Les macros suivantes ne sont pas valides: %s"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 18]
```diff
@@ -18273,3 +18273,23 @@ msgstr "Vous n'êtes pas autorisé à lister les hôtes"
...
+
+msgid "You are not allowed to add a command"
+msgstr "Vous n'êtes pas autorisé à créer une commande"
+
+msgid "Error while adding the command"
+msgstr "Erreur lors de l'ajout d'une commande"
+
+msgid "Error while retrieving a command"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18294]: a ou the? (a in EN, la in FR)

File: centreon/src/Core/Command/Application/Exception/CommandException.php [hunk 19]
```diff
@@ -42,4 +44,100 @@ public static function errorWhileSearching(\Throwable $ex): self
...
+
+    /**
+     * @param string $name
+     *
+     * @return CommandException
+     */
+    public static function nameAlreadyExists(string $name): self
+    {
+        return new self(
+            sprintf( _("'%s' command name already exists"), $name),
```
Comment (cg-tw) [inline centreon/src/Core/Command/Application/Exception/CommandException.php:56]:
```suggestion
            sprintf( _("The '%s' command name already exists"), $name),
```

File: centreon/src/Core/Command/Application/Exception/CommandException.php [hunk 20]
```diff
@@ -42,4 +44,100 @@ public static function errorWhileSearching(\Throwable $ex): self
...
+
+    /**
+     * @param string[] $arguments
+     *
+     * @return CommandException
+     */
+    public static function invalidArguments(array $arguments): self
+    {
+        return new self(
+            sprintf( _('Following arguments are not valid: %s'), implode(', ', $arguments)),
```
Comment (cg-tw) [inline centreon/src/Core/Command/Application/Exception/CommandException.php:82]:
```suggestion
            sprintf( _('The following arguments are not valid: %s'), implode(', ', $arguments)),
```

File: centreon/src/Core/Command/Application/Exception/CommandException.php [hunk 21]
```diff
@@ -42,4 +44,100 @@ public static function errorWhileSearching(\Throwable $ex): self
...
+
+    /**
+     * @param string[] $macros
+     *
+     * @return CommandException
+     */
+    public static function invalidMacros(array $macros): self
+    {
+        return new self(
+            sprintf( _('Following macros are not valid: %s'), implode(', ', $macros)),
```
Comment (cg-tw) [inline centreon/src/Core/Command/Application/Exception/CommandException.php:95]:
```suggestion
            sprintf( _('The following macros are not valid: %s'), implode(', ', $macros)),
```

File: centreon/src/Core/Command/Infrastructure/Repository/DbReadCommandRepository.php [hunk 22]
```diff
@@ -183,13 +255,104 @@ public function findByRequestParameterAndTypes(
...
+     *
+     * @throws \Throwable
+     *
+     * @return CommandMacro[]
+     */
+    private function findMacrosByCommandId(int $commandId): array
+    {
+        $statement = $this->db->prepare($this->translateDbName(
+            <<<'SQL'
+                SELECT command_macro_name, command_macro_desciption, command_macro_type
```
Comment (cg-tw) [inline centreon/src/Core/Command/Infrastructure/Repository/DbReadCommandRepository.php:330]:
```suggestion
                SELECT command_macro_name, command_macro_description, command_macro_type
```

File: centreon/src/Core/Command/Infrastructure/Repository/DbReadCommandRepository.php [hunk 23]
```diff
@@ -183,13 +255,104 @@ public function findByRequestParameterAndTypes(
...
+         *      command_macro_type: string
+         *  } $result
+         */
+        foreach ($statement as $result) {
+            $macro = new CommandMacro(
+                commandId: $commandId,
+                type: CommandMacroType::from($result['command_macro_type']),
+                name: $result['command_macro_name']
+            );
+            $macro->setDescription($result['command_macro_desciption'] ?? '');
```
Comment (cg-tw) [inline centreon/src/Core/Command/Infrastructure/Repository/DbReadCommandRepository.php:352]:
```suggestion
            $macro->setDescription($result['command_macro_description'] ?? '');
```

File: centreon/src/Core/Command/Infrastructure/Repository/DbWriteCommandRepository.php [hunk 24]
```diff
@@ -0,0 +1,202 @@
...
+            $this->debug("No macro for command {$commandId}");
+
+            return;
+        }
+
+        $request = <<<'SQL'
+            INSERT INTO `:db`.on_demand_macro_command
+            (
+                command_macro_name,
+                command_macro_desciption,
```
Comment (cg-tw) [inline centreon/src/Core/Command/Infrastructure/Repository/DbWriteCommandRepository.php:174]:
```suggestion
                command_macro_description,
```

File: centreon/tests/php/Core/Command/Application/UseCase/AddCommand/AddCommandTest.php [hunk 25]
```diff
@@ -0,0 +1,278 @@
...
+    ($this->useCase)($this->request, $this->presenter);
+
+    expect($this->presenter->response)
+        ->toBeInstanceOf(ForbiddenResponse::class)
+        ->and($this->presenter->response->getMessage())
+        ->toBe(CommandException::addNotAllowed()->getMessage());
+});
+
+it(
+    'should present an ConflictResponse when an a request parameter is invalid',
```
Comment (cg-tw) [inline centreon/tests/php/Core/Command/Application/UseCase/AddCommand/AddCommandTest.php:154]:
```suggestion
    'should present a ConflictResponse when an a request parameter is invalid',
```

File: centreon/tests/php/Core/Command/Application/UseCase/AddCommand/AddCommandTest.php [hunk 26]
```diff
@@ -0,0 +1,278 @@
...
+
+        expect($this->presenter->response)
+            ->toBeInstanceOf(ErrorResponse::class)
+            ->and($this->presenter->response->getMessage())
+            ->toBe(CommandException::errorWhileAdding(new \Exception())->getMessage());
+    }
+);
+
+it(
+    'should present a AddCommandResponse when everything has gone well',
```
Comment (cg-tw) [inline centreon/tests/php/Core/Command/Application/UseCase/AddCommand/AddCommandTest.php:218]:
```suggestion
    'should present an AddCommandResponse when everything has gone well',
```

File: centreon/tests/php/Core/Command/Domain/Model/CommandTest.php [hunk 27]
```diff
@@ -0,0 +1,182 @@
...
+    \Assert\InvalidArgumentException::class,
+    AssertionException::maxLength(
+        str_repeat('a', Command::NAME_MAX_LENGTH + 1),
+        Command::NAME_MAX_LENGTH + 1,
+        Command::NAME_MAX_LENGTH,
+        'Command::name'
+    )->getMessage()
+);
+
+it('should throw an exception when name contains invalid character', function (): void {
```
Comment (cg-tw) [inline centreon/tests/php/Core/Command/Domain/Model/CommandTest.php:121]:
```suggestion
it('should throw an exception when name contains invalid characters', function (): void {
```

File: centreon/tests/php/Core/Command/Domain/Model/NewCommandTest.php [hunk 28]
```diff
@@ -0,0 +1,168 @@
...
+    \Assert\InvalidArgumentException::class,
+    AssertionException::maxLength(
+        str_repeat('a', NewCommand::NAME_MAX_LENGTH + 1),
+        NewCommand::NAME_MAX_LENGTH + 1,
+        NewCommand::NAME_MAX_LENGTH,
+        'NewCommand::name'
+    )->getMessage()
+);
+
+it('should throw an exception when name contains invalid character', function (): void {
```
Comment (cg-tw) [inline centreon/tests/php/Core/Command/Domain/Model/NewCommandTest.php:86]:
```suggestion
it('should throw an exception when name contains invalid characters', function (): void {
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 29]
```diff
@@ -18273,3 +18273,23 @@ msgstr "Vous n'êtes pas autorisé à lister les hôtes"
...
+msgstr "Erreur lors de la récuparation de la commande"
```
Comment (a-launois) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18295]:
```suggestion
msgstr "Erreur lors de la récuparation d'une commande"
```

File: centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 30]
```diff
@@ -16524,3 +16524,23 @@ msgstr ""
...
+#  msgstr ""
+
+#  msgid "'%d' is not a valid command type"
+#  msgstr ""
+
+#  msgid "Following arguments are not valid: %s"
+#  msgstr ""
+
+#  msgid "Following macros are not valid: %s"
+#  msgstr ""
```
Comment (a-launois) [inline centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po:16537]:
```suggestion
#  msgid "The '%s' command name already exists"
#  msgstr ""

#  msgid "'%d' is not a valid command type"
#  msgstr ""

#  msgid "The following arguments are not valid: %s"
#  msgstr ""

#  msgid "The following macros are not valid: %s"
#  msgstr ""
```

File: centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po [hunk 31]
```diff
@@ -16188,3 +16188,24 @@ msgstr ""
...
+#  msgstr ""
+
+#  msgid "'%d' is not a valid command type"
+#  msgstr ""
+
+#  msgid "Following arguments are not valid: %s"
+#  msgstr ""
+
+#  msgid "Following macros are not valid: %s"
+#  msgstr ""
```
Comment (a-launois) [inline centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po:16202]:
```suggestion
#  msgid "The '%s' command name already exists"
#  msgstr ""

#  msgid "'%d' is not a valid command type"
#  msgstr ""

#  msgid "The following arguments are not valid: %s"
#  msgstr ""

#  msgid "The following macros are not valid: %s"
#  msgstr ""
```

File: centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 32]
```diff
@@ -16989,3 +16989,23 @@ msgstr ""
...
+#  msgstr ""
+
+#  msgid "'%d' is not a valid command type"
+#  msgstr ""
+
+#  msgid "Following arguments are not valid: %s"
+#  msgstr ""
+
+#  msgid "Following macros are not valid: %s"
+#  msgstr ""
```
Comment (a-launois) [inline centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:17002]:
```suggestion
#  msgid "The '%s' command name already exists"
#  msgstr ""

#  msgid "'%d' is not a valid command type"
#  msgstr ""

#  msgid "The following arguments are not valid: %s"
#  msgstr ""

#  msgid "The following macros are not valid: %s"
#  msgstr ""
```

File: centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandResponse.yaml [hunk 33]
```diff
@@ -24,11 +24,11 @@ properties:
   is_shell:
     type: boolean
     description: |
-      If your command requires shell features like pipes, redirections, globbing etc. check this box.
+      Is required if your command requires shell features like pipes, redirections, globbing etc.
 
       If you are using the monitoring engine this option cannot be disabled.
 
-      Note that commands that require shell are slowing down the poller server.
+      Note that commands that require the features are slowing down the poller server.
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandResponse.yaml:31]: shell features?

File: centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandResponse.yaml [hunk 34]
```diff
@@ -0,0 +1,110 @@
...
+    type: string
+    description: "Command line to be executed by the poller"
+  is_shell:
+    type: boolean
+    description: |
+      Is required if your command requires shell features like pipes, redirections, globbing etc.
+
+      If you are using the monitoring engine this option cannot be disabled.
+
+      Note that commands that require the features are slowing down the poller server.
```
Comment (a-launois) [inline centreon/doc/API/v23.10/Configuration/Command/Schema/AddCommandResponse.yaml:31]:
```suggestion
      Note that commands that require the shell features are slowing down the poller server.
```

**Review by cg-tw — DISMISSED** [review]

**Review by dmyios — DISMISSED** [review]

File: centreon/src/Core/Command/Application/UseCase/AddCommand/AddCommandValidation.php [hunk 35]
```diff
@@ -0,0 +1,136 @@
...
+        $invalidArguments = [];
+        foreach ($argumentNames as $argumentName) {
+            if (! in_array($argumentName, $commandArguments, true)) {
+                $invalidArguments[] = $argumentName;
+            }
+        }
+
+        if ($invalidArguments !== []) {
+            throw CommandException::invalidArguments($invalidArguments);
+        }
```
Comment (callapa) [inline centreon/src/Core/Command/Application/UseCase/AddCommand/AddCommandValidation.php:78]: Could you describe the purpose of this control

File: centreon/src/Core/Command/Application/UseCase/AddCommand/AddCommandValidation.php [hunk 36]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    /**
+     * @param AddCommandRequest $request
+     *
+     * @throws \Throwable
+     */
+    public function assertAreValidMacros(AddCommandRequest $request): void
```
Comment (callapa) [inline centreon/src/Core/Command/Application/UseCase/AddCommand/AddCommandValidation.php:86]: Could you describe the purpose of this control?

File: centreon/src/Core/Command/Application/UseCase/AddCommand/AddCommandValidation.php [hunk 37]
```diff
@@ -0,0 +1,136 @@
...
+            throw CommandException::nameAlreadyExists($request->name);
+        }
+    }
+
+    /**
+     * @param AddCommandRequest $request
+     *
+     * @throws \Throwable
+     */
+    public function assertAreValidArguments(AddCommandRequest $request): void
```
Comment (callapa) [inline centreon/src/Core/Command/Application/UseCase/AddCommand/AddCommandValidation.php:62]: Could you describe the purpose of this control?

File: centreon/src/Core/Command/Infrastructure/Repository/DbReadCommandRepository.php [hunk 38]
```diff
@@ -183,13 +255,104 @@ public function findByRequestParameterAndTypes(
...
+            type: CommandTypeConverter::fromInt((int) $data['command_type']),
+            isShellEnabled: (bool) $data['enable_shell'],
+            isActivated: $data['command_activate'] === '1',
+            isLocked: (bool) $data['command_locked'],
+            argumentExample: $data['command_example'] ?? '',
+            connector: isset($data['connector_id']) && isset($data['connector_name'])
+                ? new SimpleEntity(
+                    id: $data['connector_id'],
+                    name: new TrimmedString($data['connector_name']),
+                    objectName: 'connector'
```
Comment (callapa) [inline centreon/src/Core/Command/Infrastructure/Repository/DbReadCommandRepository.php:270]:
```suggestion
                    objectName: 'Connector'
```

File: centreon/src/Core/Command/Infrastructure/Repository/DbReadCommandRepository.php [hunk 39]
```diff
@@ -183,13 +255,104 @@ public function findByRequestParameterAndTypes(
...
+                )
+                : null,
+            graphTemplate: isset($data['graph_template_id']) && isset($data['graph_template_name'])
+                ? new SimpleEntity(
+                    id: $data['graph_template_id'],
+                    name: new TrimmedString($data['graph_template_name']),
+                    objectName: 'graphTemplate'
```
Comment (callapa) [inline centreon/src/Core/Command/Infrastructure/Repository/DbReadCommandRepository.php:277]:
```suggestion
                    objectName: 'GraphTemplate'
```

File: centreon/src/Core/Command/Infrastructure/Repository/DbWriteCommandRepository.php [hunk 40]
```diff
@@ -0,0 +1,202 @@
...
+        try {
+            $commandId = $this->addCommand($command);
+            $this->addArguments($commandId, $command);
+            $this->addMacros($commandId, $command);
+
+            if (! $alreadyInTransaction) {
+                $this->db->commit();
+            }
+
+            $this->debug('Command added with ID ' . $commandId);
```
Comment (callapa) [inline centreon/src/Core/Command/Infrastructure/Repository/DbWriteCommandRepository.php:67]: The logs must be in the use case, not in the repository.

File: centreon/src/Core/Command/Infrastructure/Repository/DbWriteCommandRepository.php [hunk 41]
```diff
@@ -0,0 +1,202 @@
...
+
+        foreach ($command->getArguments() as $key => $argument) {
+            $statement->bindValue(":argName_{$key}", $argument->getName(), \PDO::PARAM_STR);
+            $statement->bindValue(":argDescription_{$key}", $argument->getDescription(), \PDO::PARAM_STR);
+        }
+        $statement->bindValue(':commandId', $commandId, \PDO::PARAM_STR);
+        $statement->execute();
+    }
+
+    private function addMacros(int $commandId, NewCommand $command): void
```
Comment (callapa) [inline centreon/src/Core/Command/Infrastructure/Repository/DbWriteCommandRepository.php:162]: No phpDoc

File: centreon/src/Core/GraphTemplate/Infrastructure/Repository/DbReadGraphTemplateRepository.php [hunk 42]
```diff
@@ -0,0 +1,61 @@
...
+namespace Core\GraphTemplate\Infrastructure\Repository;
+
+use Centreon\Domain\Log\LoggerTrait;
+use Centreon\Infrastructure\DatabaseConnection;
+use Core\Common\Infrastructure\Repository\AbstractRepositoryRDB;
+use Core\GraphTemplate\Application\Repository\ReadGraphTemplateRepositoryInterface;
+
+class DbReadGraphTemplateRepository extends AbstractRepositoryRDB implements ReadGraphTemplateRepositoryInterface
+{
+    use LoggerTrait;
```
Comment (callapa) [inline centreon/src/Core/GraphTemplate/Infrastructure/Repository/DbReadGraphTemplateRepository.php:33]: Useless
```suggestion
```

**Review by callapa — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Pull request merged**